src/main/java/org/truffleruby/interop/RubyMessageResolution.java

```diff
@@ -278,4 +278,15 @@ protected Object access(VirtualFrame frame, DynamicObject receiver, String name,
 
     }
 
+    @Resolve(message = "NEW")
+    public static abstract class ForeignNewNode extends Node {
+
+        @Child private DispatchHeadNode dispatchHeadNode = insert(new DispatchHeadNode(true, false, MissingBehavior.CALL_METHOD_MISSING, DispatchAction.CALL_METHOD));
```

Comment:
I'll fix these - except for viability, which could be the wrong thing to do I agree, but at least it is currently actually documented behaviour "Interop ignores visibility entirely", so we should change it separately rather than tagging it onto this.